pypoptools/pypoptesting/framework/test_framework.py

```diff
@@ -114,62 +129,40 @@ def setup(self, parent):
         self.skip_test_if_missing_module()
         self.success = TestStatus.PASSED
 
-    def sync_blocks(self, nodes=None, **kwargs):
-        sync_blocks(nodes or self.nodes, **kwargs)
-
-    def sync_pop_mempools(self, nodes=None, **kwargs):
-        sync_pop_mempools(nodes or self.nodes, **kwargs)
-
-    def sync_pop_tips(self, nodes=None, **kwargs):
-        sync_pop_tips(nodes or self.nodes, **kwargs)
-
-    def sync_all(self, nodes=None, **kwargs):
-        sync_all(nodes or self.nodes, **kwargs)
+    def _create_nodes(self, create_node: Callable[[int, pathlib.Path], Node]):
+        for i in range(self.num_nodes):
+            datadir = pathlib.Path(self.dir, "node" + str(i))
+            datadir.mkdir()
+            self.nodes.append(create_node(i, datadir))
 
-    def setup_nodes(self):
-        """"Override this method to customize the node setup"""
+    def _start_nodes(self):
```

Comment:
not every test needs to start all nodes. leave this method (setup_nodes) unchanged please

Reply:
fixed